Clean up the class properties in TypingTest

clayreimann opened the pull request:
* Rename the TypingTest component from Main

vimark1 commented:
Hey @clayreimann  - can we get it up to date with master and merge? if we decide on not using Arrow Function in Class Property, perhaps we can add an eslint rule for it?

clayreimann commented:
I'll update this this evening. I didn't see an ESLint rule for this; I'm also less opposed to this notation now, I thought class-properties was still a stage-1/2 proposal. Given that it's actually stage-3 with some implementations available in browser I'll reconsider this. However, not all of the arrow functions here actually need `this` so I'll leave those as regular methods.

clayreimann commented:
@vimark1 I've rebased on master. I also cleaned things up a bit, include fixing a couple bugs. Each of the last 4 commits addresses an issue individually
